### PR TITLE
Feature/data validation

### PR DESCRIPTION
**Pull request by ReamFOX**

### CODE REVIEW

File: src/main/java/com/farion/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    private String getErrorMessage(ObjectError e) {
+        if (e instanceof FieldError) {
+            String field = ((FieldError) e).getField();
+            String message = e.getDefaultMessage();
+            return field + " " + message;
+        }
+        return e.getDefaultMessage();
+    }
```
Comment (andrii-hoienko) [inline src/main/java/com/farion/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: Could you please add some methods to hadle specific exception that you throw

File: src/main/java/com/farion/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors()
+                .stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (andrii-hoienko) [inline src/main/java/com/farion/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: Move string literals to constants

File: src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
 
 @Data
 public class CreateBookRequestDto {
+    @NotEmpty
+    @Size(max = 255, message = "can`t be longer than 255 characters")
     private String title;
+    @NotEmpty
+    @Size(max = 255, message = "can`t be longer than 255 characters")
     private String author;
+    @ISBN(message = "invalid format")
```
Comment (andrii-hoienko) [inline src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    @ISBN(message = "invalid isbn format")
```

File: src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
     private String isbn;
+    @NotNull
+    @DecimalMin(value = "0.00", inclusive = false, message = "can't be 0 or less")
     private BigDecimal price;
+    @Size(max = 255, message = "can`t be longer than 255 characters")
```
Comment (andrii-hoienko) [inline src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    @Size(max = 255, message = "Description can`t be longer than 255 characters")
```

Comment (andrii-hoienko) [reply src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java]: Make same changes where possible

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job! Left minor comments

File: src/main/java/com/farion/onlinebookstore/controller/BookController.java [hunk 5]
```diff
@@ -17,39 +21,48 @@
...
     @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @Operation(summary = "Soft delete book by id", description = "Soft delete book by specific id")
     public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update book by id", description = "Update book by specific id")
     public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
```
Comment (yevheniisynytsia) [inline src/main/java/com/farion/onlinebookstore/controller/BookController.java]:
```suggestion
    public BookDto updateById(@PathVariable Long id, @RequestBody @Valid CreateBookRequestDto requestDto) {
```

File: src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -1,14 +1,30 @@
...
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.validator.constraints.ISBN;
+import org.hibernate.validator.constraints.URL;
 
 @Data
 public class CreateBookRequestDto {
+    @NotEmpty
```
Comment (yevheniisynytsia) [inline src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java]: let's use `@NotBlank`
https://www.baeldung.com/java-bean-validation-not-null-empty-blank

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]

File: src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -11,10 +11,10 @@
...
 public class CreateBookRequestDto {
-    @NotEmpty
+    @NotBlank
     @Size(max = 255, message = "Title can`t be longer than 255 characters")
     private String title;
-    @NotEmpty
+    @NotBlank
     @Size(max = 255, message = "Author can`t be longer than 255 characters")
     private String author;
     @ISBN(message = "invalid isbn format")
```
Comment (yevheniisynytsia) [inline src/main/java/com/farion/onlinebookstore/dto/CreateBookRequestDto.java:21]: `@ISBN` accepts nulls, consider adding validation for it, you can use `@NotBlank` as well

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]

**Pull request merged**